atomic_reactor/plugin.py

```diff
@@ -135,22 +122,39 @@ def args_from_user_params(user_params: dict, /) -> dict:
         return {}
 
 
+# Built-in plugins
+class PreBuildSleepPlugin(Plugin):
+    """
+    Sleep for a specified number of seconds.
+
+    This plugin is only intended to be used for debugging.
+    """
+
+    key = 'pre_sleep'
+
+    def __init__(self, workflow, seconds=60):
+        self.seconds = seconds
+
+    def run(self):
+        time.sleep(self.seconds)
+
+
 class PluginsRunner(object):
 
-    def __init__(self, plugin_class_name, plugins_conf, *args, **kwargs):
-        """
-        constructor
+    def __init__(self, plugins_conf: List[Dict[str, Any]], *args, **kwargs) -> None:
+        """constructor
 
-        :param plugin_class_name: str, name of plugin class to filter (e.g. 'PreBuildPlugin')
-        :param plugins_conf: list of dicts, configuration for plugins
+        :param plugins_conf: list of dicts, configuration for plugins,
+            e.g. {'name': 'plugin_a', 'required': True}
+        :type plugins_conf: dict[str, any]
```

Comment:
I think it depends on how we use the docstring, and from my point of view, the types in docstring are for different purpose than the type annotation. If we would like to keep docstring to describe arguments, for the completeness, it would be good to include the type as well.

Reply:
Seems reasonable. The type in the docstring does need to be fixed though (`list[dict[str, any]]`)

Reply:
Oh, sorry, I'll update it later.